matchms/similarity/BaseSimilarity.py

```diff
@@ -28,87 +30,127 @@ class BaseSimilarity:
+            it is recommended to use `matrix()` and then convert the dense matrix to COO format.
+          - When a mask is provided, only the pairs specified in the mask are computed.
+
+        Parameters
+        ----------
+        references:
+            A collection of reference spectra.
+        queries:
+            A collection of query spectra.
+        mask_indices:
+            A COOIndex instance specifying the (row, column) pairs to compute. Defaults to None.
+        is_symmetric:
+            If True, assumes that the matrix is symmetric. Defaults to False.
+        """
         if len(self.score_filters) == 0 and mask_indices is not None:
             return self._sparse_array_with_mask_without_filter(references, queries, mask_indices=mask_indices)
         if len(self.score_filters) != 0 and mask_indices is None:
             return self._sparse_array_with_filter_without_mask(references, queries, is_symmetric=is_symmetric)
         if len(self.score_filters) != 0 and mask_indices is not None:
             return self._sparse_array_with_filter(references, queries, mask_indices)
 
-        # if score_filters is None and mask_indices is None:
-        # todo replace with using matrix followed by a conversion to COO array. (and a warning that this is not a
-        #  good idea) do this once we settle on a COO Array format (e.g. using the sparse package)
+        # TODO: replace with matrix computation followed by a conversion to COO array.
+        # (and a warning that this is not a good idea) do this once we settle on a COO Array format (e.g. using the sparse package)
         raise ValueError("If no masking or score filters is needed, please use matrix() instead")
 
-    def _matrix_without_mask_with_filter(self,
-                          references: Sequence[SpectrumType], queries: Sequence[SpectrumType],
-                          is_symmetric: bool = False):
+    # --- Dense Matrix Computations ---
+
+    def _matrix_without_mask_with_filter(
+            self,
+            references: Sequence[SpectrumType],
+            queries: Sequence[SpectrumType],
+            is_symmetric: bool = False) -> np.ndarray:
+        """
+        Compute a dense similarity matrix without a mask, then apply score filters (if any).
+
+        This method is used internally by `matrix()` when no mask is provided.
+        """
         sim_matrix = self._matrix_without_mask_without_filter(references, queries, is_symmetric=is_symmetric)
         for score_filter in self.score_filters:
             sim_matrix = score_filter.filter_matrix(sim_matrix)
         return sim_matrix
 
-    def _matrix_without_mask_without_filter(self,
-                                            references: Sequence[SpectrumType], queries: Sequence[SpectrumType],
-                                            is_symmetric: bool = False
-                                            ) -> np.ndarray:
+    def _matrix_without_mask_without_filter(
```

Comment:
We could also merge this function with _matrix_without_mask_with_filter. Would that make more sense? Or would that merge to much logic in one function?